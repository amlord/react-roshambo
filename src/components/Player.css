.Player {
    vertical-align: top;
    font-size: 1rem;
    display: inline-block;
    width: 31.578947368421053%;
    margin-right: 2.631578947368421%;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
}

.Player__title {
    font-size: 1em;
    margin: 0;
    padding: 5px;
    background: #ececed;
}

.Player__choice {
    height: 238px;
    margin: 10px;
    border-radius: 4px;
    background: #f2f2f2;

    display: flex;
    align-items: center;
    justify-content: center;
}

.Player__ChosenShape {
    margin: 0;
}