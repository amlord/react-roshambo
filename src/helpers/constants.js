export const ROCK = 1;
export const PAPER = 2;
export const SCISSORS = 3;

export const WIN = 'win';
export const DRAW = 'draw';
export const LOSS = 'loss';
