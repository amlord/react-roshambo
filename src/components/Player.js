import React, { Component } from 'react';

import './Player.css';
import { ROCK, PAPER, SCISSORS, Translate } from '../helpers/Constants';

class Player extends Component {
  render() {
    return (
      <section className="Player">
        <header>
          <h1 className="Player__title">Player</h1>
        </header>
        <div className="Player__choice">
          <h2 className="Player__ChosenShape">{Translate.choice(this.props.chosen)}</h2>
        </div>
        <button className="btn--rock" onClick={() => this.props.playerChoice(ROCK)}>Rock</button>
        <button className="btn--paper" onClick={() => this.props.playerChoice(PAPER)}>Paper</button>
        <button className="btn--scissors" onClick={() => this.props.playerChoice(SCISSORS)}>Scissors</button>
      </section>
    );
  }
}

export default Player;
