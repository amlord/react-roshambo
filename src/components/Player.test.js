import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import Player from './Player';
import { ROCK, PAPER, SCISSORS } from '../helpers/Constants';

describe('Player', () => {
  const mockplayerChoice = jest.fn();
  const props = { playerChoice: mockplayerChoice };
  const player = shallow(<Player {...props} />);

  it('renders correctly', () => {
    expect(player).toMatchSnapshot();
  });

  describe('when clicking the `rock` button', () => {
    beforeEach( () => {
      player.find('.btn--rock').simulate('click');
    });

    it('calls the `playerChoice` callback', () => {
      expect(mockplayerChoice).toHaveBeenCalledWith(ROCK);
    });
  });

  describe('when clicking the `paper` button', () => {
    beforeEach( () => {
      player.find('.btn--paper').simulate('click');
    });

    it('calls the `playerChoice` callback', () => {
      expect(mockplayerChoice).toHaveBeenCalledWith(PAPER);
    });
  });

  describe('when clicking the `scissors` button', () => {
    beforeEach( () => {
      player.find('.btn--scissors').simulate('click');
    });

    it('calls the `playerChoice` callback', () => {
      expect(mockplayerChoice).toHaveBeenCalledWith(SCISSORS);
    });
  });
});