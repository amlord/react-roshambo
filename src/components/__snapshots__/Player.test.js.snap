// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Player
    playerChoice={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1
                    className="Player__title"
          >
                    Player
          </h1>
</header>,
        <div
          className="Player__choice"
>
          <h2
                    className="Player__ChosenShape"
          />
</div>,
        <button
          className="btn--rock"
          onClick={[Function]}
>
          Rock
</button>,
        <button
          className="btn--paper"
          onClick={[Function]}
>
          Paper
</button>,
        <button
          className="btn--scissors"
          onClick={[Function]}
>
          Scissors
</button>,
      ],
      "className": "Player",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="Player__title"
>
            Player
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Player",
            "className": "Player__title",
          },
          "ref": null,
          "rendered": "Player",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2
            className="Player__ChosenShape"
/>,
          "className": "Player__choice",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "Player__ChosenShape",
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rock",
          "className": "btn--rock",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Rock",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Paper",
          "className": "btn--paper",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Paper",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Scissors",
          "className": "btn--scissors",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Scissors",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1
                        className="Player__title"
            >
                        Player
            </h1>
</header>,
          <div
            className="Player__choice"
>
            <h2
                        className="Player__ChosenShape"
            />
</div>,
          <button
            className="btn--rock"
            onClick={[Function]}
>
            Rock
</button>,
          <button
            className="btn--paper"
            onClick={[Function]}
>
            Paper
</button>,
          <button
            className="btn--scissors"
            onClick={[Function]}
>
            Scissors
</button>,
        ],
        "className": "Player",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="Player__title"
>
              Player
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Player",
              "className": "Player__title",
            },
            "ref": null,
            "rendered": "Player",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2
              className="Player__ChosenShape"
/>,
            "className": "Player__choice",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "Player__ChosenShape",
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rock",
            "className": "btn--rock",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Rock",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Paper",
            "className": "btn--paper",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Paper",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Scissors",
            "className": "btn--scissors",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Scissors",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
